### PR TITLE
Enhance agnhost netexec for SSRF E2Es

### PR DESCRIPTION
**Pull request by tallclair**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

This adds 3 pieces of functionality that will be used in an upcoming PR to add regression tests for some recent fixes for SSRF vulnerabilities:

1. Allow setting a custom response status code with the `echo` endpoint
2. Add a new `redirect` endpoint that responds with a redirect to the desired location
3. Add an option to override the URL and always use the specified handler

```release-note
NONE
```

/area test

### CODE REVIEW

File: test/images/agnhost/netexec/netexec.go [hunk 1]
```diff
@@ -112,6 +115,7 @@ func init() {
 		"File containing an x509 private key matching --tls-cert-file")
 	CmdNetexec.Flags().IntVar(&udpPort, "udp-port", 8081, "UDP Listen Port")
 	CmdNetexec.Flags().IntVar(&sctpPort, "sctp-port", -1, "SCTP Listen Port")
+	CmdNetexec.Flags().StringVar(&httpOverride, "http-override", "", "Override the HTTP handler to always respond as if it were a GET with this path & params")
```
Comment (claudiubelu) [inline test/images/agnhost/netexec/netexec.go:118]: Could you also add some information about this flag in the agnhost README file? The usage should get updated in the readme as well.

Comment (tallclair) [reply test/images/agnhost/netexec/netexec.go:125]: Done. I also noticed the TLS flags were missing from the usage string and added them there.

**Comment by claudiubelu** [comment]
The milestone should probably be updated for this to merge.

**Comment by sayanchowdhury** [comment]
👋🏽 Hi @tallclair ! I'm from the Bug Triage Team! I see that the PR was approved earlier. The code freeze for 1.20 release is starting November 12th which is about 3 weeks from now.

As the PR is tagged for 1.20, is it still planned for this release?

**Comment by tallclair** [comment]
trivial rebase, retagging.

**Pull request merged**